ts/features/wallet/component/features/PaymentStatusSwitch.tsx

```diff
@@ -0,0 +1,118 @@
+import { none, Option, some } from "fp-ts/lib/Option";
+import * as pot from "italia-ts-commons/lib/pot";
+import { View } from "native-base";
+import * as React from "react";
+import { useEffect, useState } from "react";
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import { RemoteSwitch } from "../../../../components/core/selection/RemoteSwitch";
+import { IOColors } from "../../../../components/core/variables/IOColors";
+import { IOStyleVariables } from "../../../../components/core/variables/IOStyleVariables";
+import IconFont from "../../../../components/ui/IconFont";
+import I18n from "../../../../i18n";
+import { mixpanelTrack } from "../../../../mixpanel";
+import {
+  fetchWalletsRequestWithExpBackoff,
+  updatePaymentStatus,
+  UpdatePaymentStatusPayload
+} from "../../../../store/actions/wallet/wallets";
+import { GlobalState } from "../../../../store/reducers/types";
+import { getPaymentStatusById } from "../../../../store/reducers/wallet/wallets";
+import { PaymentMethod } from "../../../../types/pagopa";
+import { showToast } from "../../../../utils/showToast";
+
+type OwnProps = {
+  paymentMethod: PaymentMethod;
+};
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps> &
+  OwnProps;
+
+/**
+ * TypeGuard to extract the good payment status some(pot)
+ * @param p
+ */
+const toOptionPot = (
+  p: pot.Pot<boolean | undefined, Error>
+): Option<pot.Pot<boolean, Error>> =>
+  pot.fold(
+    p,
+    () => some(pot.none) as Option<pot.Pot<boolean, Error>>,
+    () => some(pot.noneLoading),
+    newVal => (newVal !== undefined ? some(pot.noneUpdating(newVal)) : none),
+    error => some(pot.noneError(error)),
+    value => (value !== undefined ? some(pot.some(value)) : none),
+    value => (value !== undefined ? some(pot.someLoading(value)) : none),
+    (olddValue, newValue) =>
+      olddValue !== undefined && newValue !== undefined
+        ? some(pot.someUpdating(olddValue, newValue))
+        : none,
```

Comment:
```suggestion
    (oldValue, newValue) =>
      oldValue !== undefined && newValue !== undefined
        ? some(pot.someUpdating(oldValue, newValue))
        : none,
```

Reply:
Changed in f6dd1fd1aa4f140bbaf6144e442c26da9c41fbe7